### DISCUSSION TITLE
How to have a startup run by parent process and later worker process run it.

### DISCUSSION THREAD

**Discussion by gandharvsuri** [fastapi/fastapi#6085] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.on_event("startup")
def startup_event():
    """
    Initialize FastAPI and add variables
    """
    model_path = "./model"

    if not os.path.exists(model_path): #GCP
        download_blob(gcp_bucket,gcp_model_path,model_directory,CONST_GCP_PATH)
        
    model = ORTModelForSeq2SeqLM.from_pretrained(model_path)

    app.package = {
        "model": model,
    }
```

### Description

I have a REST API, which downlaods the model from cloud and loads it in mermory. I am using worker nodes and all of them are trying to download the model. 
I just want to downlaod the model once and allow the worker nodes to load it in memory. 
Even though I have an if condition to check the if the directory exists, all the worker nodes are able to enter the if condition block.

How do I initally run only the parent process to download the blob and then let worker nodes run freely? 

### Operating System

Linux

### FastAPI Version

0.74.1

### Python Version

3.9

**Comment by iudeen** [comment]
The worker is from Uvicorn, not from FastAPI. I am not exactly sure if you do this in this way. May be have a init container that can have the model in pickle form and then your workers can load from it?🤔

**Comment by jgould22** [comment]
Another option might be to have the workers attempt to write a lock file and let the winner do the downloading and have the rest busy wait and read data after it is downloaded.

**Comment by gandharvsuri** [comment, marked as answer]
Idk how I missed this, but one can run a simple script in the dockerfile before running the uvicorn start command. That way I don't need to download the model in the startup function. 
Anyways thanks for the suggetions, and my bad I posted it on the wrong repo.
